packages/atproto_client/models/string_formats.py

```diff
@@ -343,35 +352,65 @@ def validate_datetime(v: str, _: ValidationInfo) -> str:
     Raises:
         ValueError: If datetime format is invalid
     """
-    # Must contain uppercase T and Z if used
+    # Must not have whitespace
     if v != v.strip():
         raise ValueError('Invalid datetime: no whitespace allowed')
 
     # Must contain uppercase T
-    if 'T' not in v or ('z' in v and 'Z' not in v):
+    if 'T' not in v:
         raise ValueError('Invalid datetime: must contain uppercase T separator')
 
-    # Must have seconds (HH:MM:SS)
-    time_part = v.split('T')[1]
-    if len(time_part.split(':')) != 3:
+    # Split into date and time parts
+    try:
+        time_str = v.split('T')[1]
+    except IndexError:
+        raise ValueError('Invalid datetime: invalid format') from None
+
+    # Extract the time part before any timezone
+    if 'Z' in time_str:
+        time_part = time_str.split('Z')[0]
+    elif '+' in time_str:
+        time_part = time_str.split('+')[0]
+    elif '-' in time_str:
+        # Find the last '-' in case year is negative
+        time_part = time_str.rsplit('-', 1)[0]
+    else:
+        time_part = time_str
+
+    # Check HH:MM:SS format
+    time_segments = time_part.split(':')
+    if len(time_segments) != 3:
         raise ValueError('Invalid datetime: seconds are required')
 
     # If has decimal point, must have digits after it
-    if '.' in v and not re.search(r'\.\d+', v):
+    if '.' in time_segments[2] and not re.search(r'\.\d+', time_segments[2]):
         raise ValueError('Invalid datetime: invalid fractional seconds format')
 
-    # Must match exactly timezone pattern with nothing after
+    # Must have valid timezone
     if v.endswith('-00:00'):
         raise ValueError('Invalid datetime: -00:00 timezone not allowed')
-    if not (re.match(r'.*Z$', v) or re.match(r'.*[+-]\d{2}:\d{2}$', v)):
+
+    # Must end with Z or valid timezone offset
+    if not (v.endswith('Z') or re.search(r'[+-]\d{2}:\d{2}$', v)):
         raise ValueError('Invalid datetime: must include timezone')
 
-    # Final validation
+    # Final validation using datetime.fromisoformat
     try:
-        datetime.fromisoformat(v.replace('Z', '+00:00'))
+        # Handle both Z and explicit timezone formats
+        datetime_str = v
+        if v.endswith('Z'):
+            # For Python 3.10 compatibility, normalize fractional seconds to 6 digits
+            if '.' in v:
+                base, frac = v[:-1].split('.')
+                # Pad or truncate to exactly 6 digits
+                frac = (frac + '000000')[:6]
+                datetime_str = f'{base}.{frac}+00:00'
+            else:
+                datetime_str = v[:-1] + '+00:00'
```

Comment:
it seems that 3.10 has stricter requirements on fractional seconds, so I'm normalizing to 6 digits

Reply:
do you know the link to Python`s changelog about it or something? would be great to mention it in the comment

Reply:
added in [1b31442](https://github.com/MarshalX/atproto/pull/572/commits/1b31442342cc981dcae7f6ac8046476d4bd3448d)